packages/salesforcedx-apex-replay-debugger/test/unit/states/logEntryState.test.ts

```diff
@@ -48,7 +48,9 @@ describe('LogEntry event', () => {
     expect(stackFrame.id).to.equal(0);
     expect(stackFrame.name).to.equal('');
     expect(stackFrame.line).to.equal(context.getLogLinePosition() + 1);
-    expect(stackFrame.source.name).to.equal(context.getLogFileName());
-    expect(stackFrame.source.path).to.equal(context.getLogFilePath());
+    if (stackFrame.source) {
```

Comment:
It seems to me that source should be present and if not the test should fail.

Reply:
Source is an optional parameter though

Reply:
Yes, but the original test expects it to be there (I know assuming that it is present). The original test would fail with error "cannot access property name from undefined", if it were undefined. The change eliminates the test case. 

If you remove the guard and change expect to
```
expect(stackFrame.source?.name).to.equal(context.getLogFileName());
```
The test will pass as expected when it is not defined and will fail if it is undefined, which matches the "intent" (even if unintentional) of the test.

Reply:
Oh, that makes sense. Thanks Pete!